### PR TITLE
Add Travis build with `all-features` to current (with default features)

### PR DESCRIPTION
**Pull request by hjmallon**
#26

### CODE REVIEW

**Comment by hjmallon** [comment]
This seems to work - ish. However this change could scupper it https://github.com/rust-lang/cargo/pull/7507

Maybe there is a better way to pick the features on the sub projects?

File: .travis.yml [hunk 1]
```diff
@@ -10,6 +10,9 @@ rust:
   - nightly
   - beta
   - stable
+env:
+  - CARGO_EXTRA_ARGS="--all-features"
+  - CARGO_EXTRA_ARGS=""
```
Comment (zbraniecki) [inline .travis.yml]: Is that a mistake or is it duplicated intentionally?

Comment (hjmallon) [reply .travis.yml]: Intentional so it does both. So matrix of (nightly, beta, stable) * (standard features, all features).

**Comment by zbraniecki** [comment]
this doesn't complete on travis?

**Comment by hjmallon** [comment]
It doesn't complete on Travis because:

1. Nightly blocked the --all-features flag on workspaces recently (they reverted that here https://github.com/rust-lang/cargo/pull/7525)
2. Tests don't pass with --all-features enabled. There is something failing in remove-likely-subtags. I haven't looked into it further though.

**Comment by hjmallon** [comment]
#28 PR fixes the `cargo test --all-features` failing (point 2 above).
Hopefully tonight's nightly will fix point 1 above.

**Comment by hjmallon** [comment]
Still waiting for a cargo nightly after the one below. I am not sure how to check when those happen but I'm happy to wait a few days (it should be soon?).

```sh
% cargo -V
cargo 1.40.0-nightly (3a9abe3f0 2019-10-15)
```

**Comment by hjmallon** [comment]
I am now happy with this and it is passing CI.

**Comment by zbraniecki** [comment]
excellent! Thank you!

**Pull request merged**